source/retryable-reads/tests/README.rst

```diff
@@ -0,0 +1,269 @@
+``times`` or ``skip``, which are mutually exclusive:
+
+- ``{ times: <integer> }`` may be used to limit the number of times the fail
+  point may trigger before transitioning to ``"off"``.
+- ``{ skip: <integer> }`` may be used to defer the first trigger of a fail
+  point, after which it will transition to ``"alwaysOn"``.
+
+The ``data`` option is a document that may be used to specify options that
+control the fail point's behavior. ``failCommand`` supports the following
+``data`` options, which may be combined if desired:
+
+- ``failCommands``: Required, the list of command names to fail.
+- ``closeConnection``: Boolean option, which defaults to ``false``. If
+  ``true``, the connection on which the command is executed will be closed
+  and the client will see a network error.
+- ``errorCode``: Integer option, which is unset by default. If set, the
+  specified command error code will be returned as a command error.
+
+Disabling Fail Point after Test Execution
+-----------------------------------------
+
+After each test that configures a fail point, drivers should disable the
+``failCommands`` fail point to avoid spurious failures in
+subsequent tests. The fail point may be disabled like so::
+
+    db.runCommand({
+        configureFailPoint: "failCommands",
+        mode: "off"
+    });
+
+Network Error Tests
+===================
+
+Network error tests are expressed in YAML and should be run against a standalone,
+shard cluster, or single-node replica set. Until `SERVER-34943`_ is resolved,
+these tests cannot be run against a multi-node replicaset.
+
+
+.. _SERVER-34943: https://jira.mongodb.org/browse/SERVER-34943
+
+Test Format
+-----------
+
+Test Format
+===========
+
+Each YAML file has the following keys:
+
+- ``database_name`` and ``collection_name``: Optional. The database and collection to use
+  for testing.
+
+- ``data``: The data that should exist in the collection under test before each
+  test run.
+
+- ``tests``: An array of tests that are to be run independently of each other.
+  Each test will have some or all of the following fields:
+
+  - ``description``: The name of the test.
+
+  - ``clientOptions``: Optional, parameters to pass to MongoClient().
+
+  - ``failPoint``: Optional, a server failpoint to enable expressed as the
+    configureFailPoint command to run on the admin database.
+
+  - ``sessionOptions``: Optional, parameters to pass to
+    MongoClient.startSession().
+
+  - ``operation``: A document describing an operation to be
+    executed. The document has the following fields:
+
+    - ``name``: The name of the operation on ``object``.
+
+    - ``object``: The name of the object to perform the operation on. Can be
+      "database", "collection", or "client."
+
+    - ``collectionOptions``: Optional, parameters to pass to the Collection()
+      used for this operation.
+
+    - ``command_name``: Present only when ``name`` is "runCommand". The name
```

Comment:
Is this actually relevant to these spec tests? I only see `command_name` referenced in APM expectations. Also, the spec isn't intended to pertain to `runCommand` so I'm not sure why you'd even test for that.

Reply:
You're right: command_name is not needed here, nor is it needed in the APM expectations. Removing them.